### PR TITLE
Basics/Commands: add experimental CC SDK paths option

### PR DESCRIPTION
**Pull request by MaxDesiatov**
### Motivation:

The experimental option doesn't do anything yet, but this PR lays some groundwork for cross-compilation SDK bundles support.

### Modifications:

Extended `FileSystem` to support new `sdks` subdirectory of either `idiomaticSwiftPMDirectory` or `dotSwiftPM` directories. Added new option to `LocationOptions`.

### Result:

`--experimental-cc-sdks-path` option can be passed a path to a directory with cross-compilation SDK bundles. The option is hidden and as of this PR has no effect whatsoever.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

File: Sources/Commands/Options.swift [hunk 1]
```diff
@@ -83,6 +83,9 @@ struct LocationOptions: ParsableArguments {
     /// Path to the compilation destination describing JSON file.
     @Option(name: .customLong("destination"), help: .hidden, completion: .directory)
     var customCompileDestination: AbsolutePath?
+
+    @Option(name: .customLong("experimental-cc-sdks-path"), help: .hidden, completion: .directory)
```
Comment (neonichu) [inline Sources/Commands/Options.swift:87]: I know it's a hidden option, but could we still unroll the `cc` part somehow or use a different abbreviation? This way it just sounds as if it is related to C compilation.

**Review by neonichu — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

File: Sources/CoreCommands/SwiftTool.swift [hunk 2]
```diff
@@ -777,6 +781,22 @@ private func getSharedCacheDirectory(options: GlobalOptions, fileSystem: FileSys
...
+    fileSystem: FileSystem
+) throws -> AbsolutePath? {
+    if let explicitCCSDKsDirectory = options.locations.ccSDKsDirectory {
+        // Create the explicit SDKs path if necessary
+        if !fileSystem.exists(explicitCCSDKsDirectory) {
+            try fileSystem.createDirectory(explicitCCSDKsDirectory, recursive: true)
+        }
+        return explicitCCSDKsDirectory
+    } else {
+        // further validation is done in workspace
```
Comment (tomerd) [inline Sources/CoreCommands/SwiftTool.swift:795]: nit:  this comment is probably incorrect in this case?

**Review by tomerd — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Pull request merged**